src/arch.rs

```diff
@@ -0,0 +1,95 @@
+#[derive(Clone, Debug)]
+pub enum Arch {
+    X86,
+    X64,
+    Arm64,
+    Armv7l,
+    Ppc64le,
+    Ppc64,
+    S390x,
+}
+
+#[cfg(unix)]
+/// Get a sane default architecture for the platform.
+pub fn default_str() -> &'static str {
+    use crate::system_info::{platform_arch, platform_name};
+
+    // TODO: Handle (arch, name, version) when Node v15+ supports darwin-arm64
+    match (platform_name(), platform_arch()) {
+        ("darwin", "arm64") => "x64",
+        (_, arch) => arch,
+    }
+}
+
+#[cfg(windows)]
+/// Get a sane default architecture for the platform.
+pub fn default_str() -> &'static str {
+    return crate::system_info::platform_arch();
+}
+
+pub fn get_default() -> Arch {
+    match from_str(default_str()) {
+        Ok(arch) => arch,
+        Err(e) => panic!("{}", e.details),
+    }
+}
```

Comment:
This can get more Rusty by `impl Default for Arch`. When doing this, we can also `impl ToString for Arch` for `default_str` and then we can let [structopt infer the default value by not giving a value to `default_value`](https://docs.rs/structopt/0.3.21/structopt/#default-values)

Reply:
I think that by implementing `Display` (like you do in this file later on), `ToString` is automatically implemented as well. So all we need is to implement `Default` for `Arch` and we will get the Structopt inference 😁

Reply:
0b3d190